src/t_stream.c

```diff
@@ -1841,9 +1841,11 @@ void xaddCommand(client *c) {
 
     /* Let's rewrite the ID argument with the one actually generated for
      * AOF/replication propagation. */
-    robj *idarg = createObjectFromStreamID(&id);
-    rewriteClientCommandArgument(c,idpos,idarg);
-    decrRefCount(idarg);
+    if (!parsed_args.id_given) {
```

Comment:
maybe we should compare `parsed_args.id` to `id` too?
AFAICT streamAppendItem respects the ID it was given, but maybe it could have also incremented the sequence number or something like that?
@guybe7 WDYT?